packages/connect-form/src/connection-string-input.tsx

```diff
@@ -97,27 +97,10 @@ function reducer(state: State, action: Action): State {
 export function hidePasswordInConnectionString(
   connectionString: string
 ): string {
-  try {
-    const passwordHiddenConnectionString = new ConnectionStringUrl(
-      connectionString
-    );
-
-    if (passwordHiddenConnectionString.password) {
-      passwordHiddenConnectionString.password = '*****';
-    }
-    if (passwordHiddenConnectionString.searchParams.get('AWS_SESSION_TOKEN')) {
-      passwordHiddenConnectionString.searchParams.set(
-        'AWS_SESSION_TOKEN',
-        '*****'
-      );
-    }
-
-    return passwordHiddenConnectionString.toString();
-  } catch (e) {
-    // If we cannot parse the connection string we'll return it
-    // as could be misformed.
-    return connectionString;
-  }
+  return redactConnectionString(connectionString).replace(
+    /<credentials>/g,
+    '*****'
+  );
```

Comment:
@Anemy @gribnoysup For context, we already had talked about this in https://github.com/mongodb-js/compass/pull/2544#discussion_r736238345 – this does differ a bit, because now invalid connection strings also end up being redacted. I think that’s the slightly better behavior, but I’m also happy to undo this if you prefer.

Reply:
I like this behavior a lot more! Nice work pulling these together and helping share between the projects.

Reply:
Makes sense to me too